activerecord/lib/active_record/core.rb

```diff
@@ -40,24 +40,29 @@ module Core
       #
       # ...would result in ActiveRecord::Base.configurations to look like this:
       #
-      #   {
-      #      'development' => {
-      #         'adapter'  => 'sqlite3',
-      #         'database' => 'db/development.sqlite3'
-      #      },
-      #      'production' => {
-      #         'adapter'  => 'sqlite3',
-      #         'database' => 'db/production.sqlite3'
-      #      }
-      #   }
+      #   #<ActiveRecord::DatabaseConfigurations:0x00007fd1acbdf800 @configurations=[
+      #     #<ActiveRecord::DatabaseConfigurations::HashConfig:0x00007fd1acbded10 @env_name="development",
+      #       @spec_name="primary", @config={"adapter"=>"sqlite3", "database"=>"db/development.sqlite3"}>,
+      #     #<ActiveRecord::DatabaseConfigurations::HashConfig:0x00007fd1acbdea90 @env_name="production",
+      #       @spec_name="primary", @config={"adapter"=>"mysql2", "database"=>"db/production.sqlite3"}>
+      #   ]>
       def self.configurations=(config)
-        @@configurations = ActiveRecord::ConnectionHandling::MergeAndResolveDefaultUrlConfig.new(config).resolve
+        @@configurations = ActiveRecord::DatabaseConfigurations.new(config)
       end
       self.configurations = {}
 
       # Returns fully resolved configurations hash
-      def self.configurations
-        @@configurations
+      #
+      # To use the deprecated hash version of ActiveRecord::Configurations, set
+      # ActiveRecord::Base.use_legacy_configurations to true.
+      def self.configurations(legacy: false)
+        if legacy
+          ActiveSupport::Deprecation.warn \
+            "Returning a hash of configs from ActiveRecord::Base.configurations is deprecated. In Rails 6.1 it will return an array of objects."
+          @@configurations.to_h
+        else
+          @@configurations
+        end
```

Comment:
Yeah, I'm happy to change this without a deprecation: it's a low-traffic API, and we're fundamentally rearranging enough that anyone relying on the legacy version is going to have a deeply limited view of the world, at best.

I think we should ditch the `legacy: true` option entirely, though: `ActiveRecord::Base.configurations.to_h` will work just fine, and is a better spelling for "be like the old one" because it works on the old one too -- the only reasonable reason to want the old hash style is because you're trying to support multiple versions.

Reply:
Added some method missing and removed the `legacy` stuff.